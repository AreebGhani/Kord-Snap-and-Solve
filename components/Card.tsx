import React from "react";
import { Text, View } from "react-native";

type Props = {};

const Card: React.FC<Props> = (props: Props) => {
  return (
    <View>
      <Text className="text-red-500 text-[18px] px-4 text-center">
        Open up /app/index.tsx to start working on your app!
      </Text>
    </View>
  );
};

export default Card;
