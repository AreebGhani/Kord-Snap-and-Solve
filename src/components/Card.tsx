/**
 * The Card component is a simple React component that displays a text message in a View component.
 */
import { Text, View } from 'react-native';

const Card = () => (
  <View>
    <Text className="text-red-500 text-[18px] px-4 text-center">
      Open up /app/index.tsx to start working on your app!
    </Text>
  </View>
);

export default Card;
