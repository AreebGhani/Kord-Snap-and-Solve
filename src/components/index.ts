/* This code snippet is exporting components from different files in a TypeScript project. Each line is
exporting a default component from a specific file using the `export { default as ComponentName }
from 'filePath';` syntax. */
export { default as Card } from '@/src/components/Card';
