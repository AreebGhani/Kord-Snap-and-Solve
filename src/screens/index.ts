/* This code snippet is exporting screens from different files in a TypeScript project. Each line is
exporting a default screen from a specific file using the `export { default as ScreenName }
from 'filePath';` syntax. */
export { default as HomeScreen } from '@/src/screens/HomeScreen';
