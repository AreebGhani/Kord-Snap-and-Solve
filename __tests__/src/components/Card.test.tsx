/* This code snippet is a test case written using Jest and React Test Renderer for a React component
called `Card`. Here's a breakdown of what each part does: */
import React from 'react';
import renderer from 'react-test-renderer';

import { Card } from '@/src/components';

describe('Card Component', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<Card />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
