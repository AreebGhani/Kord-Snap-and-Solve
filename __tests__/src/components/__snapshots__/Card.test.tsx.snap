// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders correctly 1`] = `
<View>
  <Text
    style={
      [
        {
          "color": "#ef4444",
        },
        {
          "fontSize": 18,
        },
        {
          "paddingLeft": 16,
          "paddingRight": 16,
        },
        {
          "textAlign": "center",
        },
      ]
    }
  >
    Open up /app/index.tsx to start working on your app!
  </Text>
</View>
`;
